### ISSUE TITLE
In some cases .load_pgn() cannot understand the output of .pgn()

### ISSUE THREAD

**Issue by pnb** [jhlywa/chess.js#311]
I encountered this issue in a recent game. PGN as outputted by chess.com:

```
1. e4 c5 2. Nc3 Nc6 3. Bc4 Nf6 4. f4 d6 5. Nf3 h6 6. O-O e5 7. d3 Be7 8. f5 O-O
9. a3 Nd7 10. Bd5 Nb6 11. Bxc6 bxc6 12. b4 c4 13. a4 cxd3 14. cxd3 Rb8 15. b5
cxb5 16. axb5 Qd7 17. Qa4 Nxa4 18. Rxa4 Qc7 19. Bb2 Bd7 20. Nd5 Qd8 21. Rxa7
Rxb5 22. Rb1 Qb8 23. Rxd7 Rxb2 24. Rxb2 Qxb2 25. Nxe7+ Kh7 26. Nd5 Qb1+ 27. Kf2
Qxd3 28. Rxd6 Qxe4 29. f6 Qc2+ 30. Kg3 g6 31. Nxe5 Qb3+ 32. Nf3 Re8 33. Ne7 Rb8
34. Nc6 Rc8 35. Nd4 Qb4 36. Rd7 Rc5 37. Rxf7+ Kg8 38. Re7 h5 39. f7+ Kf8 40.
Ne6+ Kxe7 41. f8=Q+ Kxe6 42. Qg8+ Kf5 43. Nh4+ Kg5 44. Nf3+ Kf5 45. Qf7+ Ke4 46.
Qe6+ Kd3 47. Qd6+ Kc3 48. Qd2+ Kb3 49. Nd4+ Ka4 50. Qd1+ Ka5 51. Qa1+ Kb6 52.
Ne6 Qg4+ 53. Kf2 Rc2+ 54. Kf1 Qxg2+
```

The `load_pgn()` function understands this PGN fine. However, when exporting the game again via `pgn()`, move **34. Nd4** gets converted to **34. Ncd4**, which would make sense because either knight could move to d4 _except_ that one of them is pinned by the queen, so the "c" qualifier is not needed. See board position below after **34. Nd4**:

<img width="420" alt="image" src="https://user-images.githubusercontent.com/620428/152623119-6d58b249-40cf-4388-ba7d-1c9b008c83b6.png">

My guess is that `load_pgn()` realizes this and thus fails. But this leads to a surprising situation where `game.load_pgn(game.pgn())` fails, which I expect should not be the case.

I suppose either the `load_pgn()` function could be a bit less picky, or the `pgn()` export function could be fixed.

**Comment by jhlywa** [comment]
Thanks for the report. I'll look into it this weekend.

**Comment by jhlywa** [comment]
Thanks for the report @pnb.  This was fixed in master a few months ago.  I'll push a new version to npm today that incorporates these fixes. In the mean time, you can pull the most recent code via ` npm install git+https://github.com/jhlywa/chess.js`

```js
const { Chess } = require('./chess.js') 
                                                                                                                                                  
const chess = Chess()                                                                                                                           
                                                                                                                                                  
const pgn = `1. e4 c5 2. Nc3 Nc6 3. Bc4 Nf6 4. f4 d6 5. Nf3 h6 6. O-O e5 7. d3 Be7 8. f5 O-O                                                    
9. a3 Nd7 10. Bd5 Nb6 11. Bxc6 bxc6 12. b4 c4 13. a4 cxd3 14. cxd3 Rb8 15. b5                                                                   
cxb5 16. axb5 Qd7 17. Qa4 Nxa4 18. Rxa4 Qc7 19. Bb2 Bd7 20. Nd5 Qd8 21. Rxa7                                                                    
Rxb5 22. Rb1 Qb8 23. Rxd7 Rxb2 24. Rxb2 Qxb2 25. Nxe7+ Kh7 26. Nd5 Qb1+ 27. Kf2                                                                 
Qxd3 28. Rxd6 Qxe4 29. f6 Qc2+ 30. Kg3 g6 31. Nxe5 Qb3+ 32. Nf3 Re8 33. Ne7 Rb8                                                                 
34. Nc6 Rc8 35. Nd4 Qb4 36. Rd7 Rc5 37. Rxf7+ Kg8 38. Re7 h5 39. f7+ Kf8 40.                                                                    
Ne6+ Kxe7 41. f8=Q+ Kxe6 42. Qg8+ Kf5 43. Nh4+ Kg5 44. Nf3+ Kf5 45. Qf7+ Ke4 46.                                                                
Qe6+ Kd3 47. Qd6+ Kc3 48. Qd2+ Kb3 49. Nd4+ Ka4 50. Qd1+ Ka5 51. Qa1+ Kb6 52.                                                                   
Ne6 Qg4+ 53. Kf2 Rc2+ 54. Kf1 Qxg2+`

// load PGN from string
console.log(chess.load_pgn(pgn)) // -> true

// verify exported PGN matches imported PGN
console.log(chess.pgn() === pgn.replace(/s+|\n/g, ' ')) // <- true

// load PGN from exported PGN
console.log(chess.load_pgn(chess.pgn())) // -> true
```

**Comment by jhlywa** [comment]
Published 0.12.1

**Comment by pnb** [comment]
Awesome, thanks a million! Apologies for not checking the repo first too, I should have thought of that.
